atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -963,14 +968,19 @@ public RowColumnRangeResult apply(CassandraClient client) throws Exception {
                                     startTs);
                             Limit limit = Limit.of(batchColumnRangeSelection.getBatchHint());
                             SlicePredicate pred = SlicePredicates.create(range, limit);
-
-                            Map<ByteBuffer, List<ColumnOrSuperColumn>> results = wrappingQueryRunner.multiget(
-                                    "getRowsColumnRange",
-                                    client,
-                                    tableRef,
-                                    wrap(rows),
-                                    pred,
-                                    readConsistencyProvider.getConsistency(tableRef));
+                            Map<ByteBuffer, List<ColumnOrSuperColumn>> results = Collections.emptyMap();
+                            try {
+                                results = wrappingQueryRunner.multiget(
+                                        "getRowsColumnRange",
+                                        client,
+                                        tableRef,
+                                        wrap(rows),
+                                        pred,
+                                        readConsistencyProvider.getConsistency(tableRef));
+                            } catch (TException e) {
+                                log.error("Query timed out for: {}", SafeArg.of("tableRef", tableRef), e);
```

Comment:
Will take a proper look tomorrow, but: here, you don't know the query timed out! It's just a TException, which _could_ be TimedOutException, or any other TException